good-enough-practices-for-scientific-computing.tex

```diff
@@ -434,32 +449,52 @@ \subsection*{Acknowledgments}
 
 \item
   \practice{2b}{Decompose programs into functions} that are no more than
-  one page (about 60 lines) long, do not
-  use global variables (constants are OK), and take no more than half
-  a dozen parameters.  The key motivation here is to fit the program
+  one page (about 60 lines) long. Functions are sections within a program
+  that do one particular thing. The syntax varies depending on programming
+  language, but generally you name the function, list the parameters, or 
+  information needed for that section of code, and list what information will 
+  come out of the function. 
+  % check definition of functions
+  % WOULD BE GOOD TO HAVE EXAMPLE OF A FUNCTION
+  Functions should take no more than about 5 or 6 input parameters
+  and shouldn't include variables used in other parts of the program 
+  (global variables).
+  % I'm not sure if we should discuss global variables. I'm not sure it's
+  % an easy concept to explain. Or better description of global variables here?
+  The key motivation here is to fit the program
   into the most limited memory of all: ours. Human short-term memory
   is famously incapable of holding more than about seven items at
   once \cite{miller1956}. If we are to understand
   what our software is doing, we must break it into chunks that obey
-  this limit, then create programs by combining these chunks.
+  this limit, then create programs by combining these chunks. Putting 
+  code into functions also makes it easier to test and troubleshoot if
+  something is going wrong. 
```

Comment:
Is it helpful to mention that functions are useful for pieces of logic that are repeated (do you copy and paste this bit of code?) or that may only happen once but that are un-helpfully ugly (would you like to NOT see the gory details all the time)?